src/main/java/uk/gov/homeoffice/digital/sas/timecard/validators/timeentry/TimeEntryValidator.java

```diff
@@ -20,6 +20,45 @@ public class TimeEntryValidator implements ConstraintValidator<TimeEntryConstrai
   @Override
   public boolean isValid(TimeEntry timeEntry, ConstraintValidatorContext context) {
 
+    if (timeEntry.getActualEndTime() != null
+        && timeEntry.getActualStartTime().after(timeEntry.getActualEndTime())) {
```

Comment:
It seems to work when I've tested it? I think the second condition is only run if the check shows that the end time is not null?

Reply:
Sorry. My mistake. I was reading `timeEntry.getActualStartTime() != null` instead of `timeEntry.getActualEndTime() != null`. And you're right the test you mentioned covers this scenario